crates/db_schema/src/impls/actor_language.rs

```diff
@@ -0,0 +1,350 @@
+    SiteLanguage::update(&conn, test_langs.clone(), site.id).unwrap();
+
+    let site_languages2 = SiteLanguage::read_local(&conn).unwrap();
+    // after update, site only has new languages
+    assert_eq!(test_langs, site_languages2);
+
+    Site::delete(&conn, site.id).unwrap();
+  }
+
+  #[test]
+  #[serial]
+  fn test_user_languages() {
+    let conn = establish_unpooled_connection();
+
+    let site = create_test_site(&conn);
+    let test_langs = test_langs1(&conn);
+    SiteLanguage::update(&conn, test_langs.clone(), site.id).unwrap();
+
+    let person_form = PersonForm {
+      name: "my test person".to_string(),
+      public_key: Some("pubkey".to_string()),
+      ..Default::default()
+    };
+    let person = Person::create(&conn, &person_form).unwrap();
+    let local_user_form = LocalUserForm {
+      person_id: Some(person.id),
+      password_encrypted: Some("my_pw".to_string()),
+      ..Default::default()
+    };
+    let local_user = LocalUser::create(&conn, &local_user_form).unwrap();
+    let local_user_langs1 = LocalUserLanguage::read(&conn, local_user.id).unwrap();
+
+    // new user should be initialized with site languages
+    assert_eq!(test_langs, local_user_langs1);
+
+    // update user languages
+    LocalUserLanguage::update(&conn, test_langs2(&conn), local_user.id).unwrap();
+    let local_user_langs2 = LocalUserLanguage::read(&conn, local_user.id).unwrap();
+    assert_eq!(2, local_user_langs2.len());
+
+    Person::delete(&conn, person.id).unwrap();
+    LocalUser::delete(&conn, local_user.id).unwrap();
+    Site::delete(&conn, site.id).unwrap();
+  }
+
+  #[test]
+  #[serial]
+  fn test_community_languages() {
+    let conn = establish_unpooled_connection();
+    let site = create_test_site(&conn);
+    let test_langs = test_langs1(&conn);
+    SiteLanguage::update(&conn, test_langs.clone(), site.id).unwrap();
+
+    let community_form = CommunityForm {
+      name: "test community".to_string(),
+      title: "test community".to_string(),
+      public_key: Some("pubkey".to_string()),
+      ..Default::default()
+    };
+    let community = Community::create(&conn, &community_form).unwrap();
+    let community_langs1 = CommunityLanguage::read(&conn, community.id).unwrap();
+    // community is initialized with site languages
+    assert_eq!(test_langs, community_langs1);
+
+    let allowed_lang1 =
+      CommunityLanguage::is_allowed_community_language(&conn, test_langs[0], community.id);
+    assert!(allowed_lang1.is_ok());
+
+    let test_langs2 = test_langs2(&conn);
+    let allowed_lang2 =
+      CommunityLanguage::is_allowed_community_language(&conn, test_langs2[0], community.id);
+    assert!(allowed_lang2.is_err());
+
+    // limit site languages to en, fi. after this, community languages should be updated to
+    // intersection of old languages (en, fr, ru) and (en, fi), which is only fi.
+    CommunityLanguage::limit_languages(&conn, vec![test_langs[0], test_langs2[0]]).unwrap();
```

Comment:
It *might* be better to use `SiteLanguage::update`, which is the higher up path, and should run this anyway, but I leave that to you.

Reply:
No this is specifically to test the limit_languages() function.